test/TestERC20Symbolic.sol

```diff
@@ -0,0 +1,32 @@
+pragma solidity ^0.5.16;
+
+import "./firefly/FireflyCheck.sol";
+import "../contracts/ERC20.sol";
+
+contract TestERC20Symbolic is FireflyCheck {
+    function _testTransfer() public {
+        ERC20 token;
+
+        // token = (magic happens here)
+
+        address from  = address(this);
+        address to    = this.firefly_newSymbolicAddress();
+        uint256 value = this.firefly_newSymbolicUint256();
+
+        require(from != to);
+
+        uint256 balFrom = token.balanceOf(from);
+        uint256 balTo   = token.balanceOf(to);
+        require(value <= balFrom);
+
+        this.firefly_assume(token.transfer(to, value));
```

Comment:
I think we should either stick to `require(...)` for all preconditions, or `this.firefly_require(...)` (instead of `this.firefly_assume`).

The reason is that Solidity developers already use `require(...)` to state function pre-conditions, so this will be more familiar. But either way we should pick one (do we use raw `require(...)` or wrapped `this.firefly_require(...)`?) and stick to it.